Indicators/RelativeVigorIndex.cs

```diff
@@ -0,0 +1,133 @@
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using QuantConnect.Data.Market;
+
+namespace QuantConnect.Indicators
+{
+    /// <summary>
+    /// The Ehlers Relative Vigor Index to be used when the symbol is in Cycle mode.  
+    /// The oscillator is basiclly in phase with the cyclic component fo the market prices.
+    /// 
+    /// It is also possible to generate a leading function if the summation window
+    /// is less than half wavelength of the dominant cycle.
+    /// 
+    /// If the dominant cycle period is not available, you can sum the RVI components
+    /// over a fixed default period.  A period of 8 or 10 is suggested because most 
+    /// cycles of interest are 16 or 20 bars long.
+    /// 
+    /// A crossing of the indicator with the indicator with a lag of 1 bar is a good way 
+    /// to create an unambiguous signal.
+    /// </summary>
+    public class RelativeVigorIndex : TradeBarIndicator
+    {
+        private RollingWindow<IndicatorDataPoint> value1;
+        private RollingWindow<IndicatorDataPoint> value2;
+        /// <summary>
+        /// 
+        /// </summary>
+        public RollingWindow<IndicatorDataPoint> RviWindow { get; set; }
+        /// <summary>
+        /// 
+        /// </summary>
+        public RollingWindow<TradeBar> Bars { get; set; }
+
+        /// <summary>
+        /// Creates the indicator
+        /// </summary>
+        /// <param name="name">string - the name of the indicator</param>
+        /// <param name="period">int - half the number of periods in the dominant market cycle. </param>
+        public RelativeVigorIndex(string name, int period)
+            : base(name)
+        {
+
+            value1 = new RollingWindow<IndicatorDataPoint>(period);
+            value2 = new RollingWindow<IndicatorDataPoint>(period);
+            RviWindow = new RollingWindow<IndicatorDataPoint>(period);
+            Bars = new RollingWindow<TradeBar>(period);
+        }
+
+
+        /// <summary>
+        /// Automatically names the indicator
+        /// </summary>
+        /// <param name="period">int - the length of computation cycle</param>
+        public RelativeVigorIndex(int period)
+            : this("RVI_" + period, period)
+        {
+        }
+
+        /// <summary>
+        /// Gets a flag indicating when this indicator is ready and fully initialized
+        /// </summary>
+        public override bool IsReady
+        {
+            get { return value1.IsReady && value2.IsReady && RviWindow.IsReady; }
+        }
+        /// <summary>
+        /// Computes the next value of this indicator from the given state
+        /// </summary>
+        /// <param name="input">The input given to the indicator</param>
+        /// <returns>A new value for this indicator</returns>
+        protected override decimal ComputeNextValue(TradeBar input)
+        {
+            // Save the bar for summing
+            Bars.Add(input);
```

Comment:
Instead of manually doing this you may want to derive from WindowIndicator<TradeBar>